### PR TITLE
Supply rid graph to NuGet

### LINKED ISSUE #3325: Supply RuntimeIdentifier graph to NuGet
Once this NuGet feature is implemented: https://github.com/NuGet/Client.Engineering/pull/96

We will need to set the appropriate property to supply the SDKs graph to NuGet, as well as remove the infrastructure that implicitly references Microsoft.NETCore.App.

### PR DESCRIPTION
**Pull request by dsplaisted**
Fix #3325

### CODE REVIEW

**Comment by dsplaisted** [comment]
This will also fix #3366

File: src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs [hunk 1]
```diff
@@ -412,19 +391,6 @@ public KnownFrameworkReference(ITaskItem item)
 
             public string Profile => _item.GetMetadata("Profile");
 
-            public string LegacyFrameworkPackages
-            {
-                get
-                {
-                    var packages = _item.GetMetadata("LegacyFrameworkPackages");
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs]: Do you have a change queued to remove this metadata from bundled versions?

**Review by nguerrera — APPROVED** [review]

**Review by peterhuene — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs (hunk 1 above)
Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs]: I now have this: https://github.com/dotnet/core-sdk/pull/2970

**Review by nkolev92 — APPROVED** [review]

**Comment by nguerrera** [comment]
Does this need a project system change? I know the shape of the interface doesn't change for properties, but do we still need to register the property somehow as being part of the nomination?

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Common.targets [hunk 2]
```diff
@@ -25,6 +25,9 @@ Copyright (c) .NET Foundation. All rights reserved.
           Should be re-visited when multiple TFM support is added to Dependencies logic.
     -->
     <DefaultImplicitPackages>Microsoft.NETCore.App;NETStandard.Library</DefaultImplicitPackages>
+
+    <!-- Set RuntimeIdentifier graph for NuGet -->
+    <RuntimeIdentifierGraphPath Condition="'$(RuntimeIdentifierGraphPath)' == ''">$(BundledRuntimeIdentifierGraphFile)</RuntimeIdentifierGraphPath>
```
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Common.targets]: With this in targets, is there a good way to disable this in the csproj? Do we need one? Suppose I want to use a newer one from a package...

**Comment by nguerrera** [comment]
I see this is already supported in the project-system: https://github.com/dotnet/project-system/pull/4952

**Comment by dsplaisted** [comment]
> Does this need a project system change? I know the shape of the interface doesn't change for properties, but do we still need to register the property somehow as being part of the nomination?

Great point.  @nkolev92 @drewnoakes @davkean Is there an updated needed for the restore nomination to include the `RuntimeIdentifierGraphPath` property?

**Comment by nguerrera** [comment]
Sorry, see my follow-up, the change is in project-system already, and 16.2 at that.

**Comment by davkean** [comment]
PR is: https://github.com/dotnet/project-system/pull/4952. We're ahead of you on this one. :)

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Common.targets (hunk 2 above)
Comment (dsplaisted) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Common.targets]: I've moved it to .props now

**Comment by dsplaisted** [comment]
@nguerrera FYI I am seeing flaky failures in the `dotnet store` tests in this PR.  For example, Microsoft.NET.Publish.Tests.GivenThatWeWantToStoreAProjectWithDependencies.It_stores_when_targeting_netcoreapp3: error NETSDK1082: There was no runtime pack for Microsoft.NETCore.App.Runtime.win-x64 available for the specified RuntimeIdentifier 'win-x64'

> F:\workspace\_work\1\s\artifacts\bin\Debug\dotnetLayout\dotnet msbuild /t:ComposeStore F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\Test\Test.csproj /p:RuntimeIdentifier=win10-x64 /p:ComposeDir=F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\o /p:ComposeWorkingDir=F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\w /p:DoNotDecorateComposeDir=true /p:CreateProfilingSymbols=false
Microsoft (R) Build Engine version 16.3.0-preview-19353-06+13c3987a5 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.
  Restore completed in 553.15 ms for F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\Test\Test.csproj.
F:\workspace\_work\1\s\artifacts\bin\Debug\dotnetLayout\sdk\3.0.100-preview8-012830\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.CrossGen.targets(297,5): error NETSDK1082: There was no runtime pack for Microsoft.NETCore.App.Runtime.win-x64 available for the specified RuntimeIdentifier 'win-x64'. [F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\Test\Test.csproj]
F:\workspace\_work\1\s\artifacts\bin\Debug\dotnetLayout\sdk\3.0.100-preview8-012830\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.CrossGen.targets(297,5): error MSB4018: The "ResolveRuntimePackAssets" task failed unexpectedly. [F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\Test\Test.csproj]
F:\workspace\_work\1\s\artifacts\bin\Debug\dotnetLayout\sdk\3.0.100-preview8-012830\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.CrossGen.targets(297,5): error MSB4018: System.ArgumentException: Message is not prefixed with NETSDK error code or error code is formatted incorrectly: Runtime list not found: data\RuntimeList.xml [F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\Test\Test.csproj]
F:\workspace\_work\1\s\artifacts\bin\Debug\dotnetLayout\sdk\3.0.100-preview8-012830\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.CrossGen.targets(297,5): error MSB4018:    at Microsoft.NET.Build.Tasks.Logger.DebugThrowMissingOrIncorrectCode(String code, String message, MessageLevel level) in /_/src/Tasks/Common/Logger.cs:line 127 [F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\Test\Test.csproj]
F:\workspace\_work\1\s\artifacts\bin\Debug\dotnetLayout\sdk\3.0.100-preview8-012830\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.CrossGen.targets(297,5): error MSB4018:    at Microsoft.NET.Build.Tasks.Logger.CreateMessage(MessageLevel level, String format, String[] args) in /_/src/Tasks/Common/Logger.cs:line 102 [F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\Test\Test.csproj]
F:\workspace\_work\1\s\artifacts\bin\Debug\dotnetLayout\sdk\3.0.100-preview8-012830\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.CrossGen.targets(297,5): error MSB4018:    at Microsoft.NET.Build.Tasks.Logger.LogError(String format, String[] args) in /_/src/Tasks/Common/Logger.cs:line 66 [F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\Test\Test.csproj]
F:\workspace\_work\1\s\artifacts\bin\Debug\dotnetLayout\sdk\3.0.100-preview8-012830\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.CrossGen.targets(297,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() in /_/src/Tasks/Common/TaskBase.cs:line 42 [F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\Test\Test.csproj]
F:\workspace\_work\1\s\artifacts\bin\Debug\dotnetLayout\sdk\3.0.100-preview8-012830\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.CrossGen.targets(297,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\Test\Test.csproj]
F:\workspace\_work\1\s\artifacts\bin\Debug\dotnetLayout\sdk\3.0.100-preview8-012830\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.CrossGen.targets(297,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [F:\workspace\_work\1\s\artifacts\tmp\Debug\testSdkPublish\w\It_stores_whe---79BFCBD3\Test\Test.csproj]
>
> Exit Code: 1

**Comment by nguerrera** [comment]
Strange, have you seen that before?

**Comment by nguerrera** [comment]
I'm having a hard time seeing how this would be flaky: https://github.com/dotnet/sdk/blob/c5fefbaf6807df7b4991c8a802beaeccf5a5df48/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.CrossGen.targets#L297

**Comment by nguerrera** [comment]
> Runtime list not found: data\RuntimeList.xml 

Hmm.

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToCopyLocalDependencies.cs [hunk 3]
```diff
@@ -330,7 +331,8 @@ public void It_does_not_copy_local_runtime_dependencies_for_netframework_project
             });
         }
 
-        [Fact]
+        //  Core MSBuild only because CI machines don't have updated VS (with support for RuntimeIdentifierGraphPath)
```
Comment (nguerrera) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToCopyLocalDependencies.cs]: File an issue to turn these back on?

**Comment by dsplaisted** [comment]
> Runtime list not found: data\RuntimeList.xml

I think this is actually an extra error message related to the same root cause as the missing runtime pack.

**Comment by dsplaisted** [comment]
@nguerrera I am seeing a similar failure for the DesignTimeBuildSucceedsAfterRuntimeIdentifierIsChanged test on full framework in #3100 and #3108

**Comment by nguerrera** [comment]
Ok so not store related?

**Comment by nguerrera** [comment]
I was able to repro a store failure and put in a targeted fix in 9f2ae05. But I think there's a general problem for the case where RuntimeIndentifier != null, SelfContained == false, RuntimeIdentifiers does not include RuntimeIdentifier. That may be what hit the other PR.

**Comment by nguerrera** [comment]
The DesignTimeBuildSucceedsAfterRuntimeIdentifierIsChanged appears to be due to turning ResolveRuntimePackAssets back on for design time builds. We change the RID, then expect the build to pass, but we haven't restored the runtime pack yet.

All of these issues appear to hide themselves behind nuget cache reuse / test order. :(

**Comment by nguerrera** [comment]
@dotnet/dotnet-cli Can some review 9f2ae05 and a7671b1 that I added to try to get this to green?

**Review by livarcocc — APPROVED** [review]

**Comment by onovotny** [comment]
With this going into master, will it be also cherry-picked for .NET Core 3?

**Comment by nguerrera** [comment]
master is still .NET Core 3 for the SDK repos. :) This will be in preview 8.

**Pull request merged**